src/cljs/athens/devcards/devtool.cljs

```diff
@@ -0,0 +1,427 @@
+
+
+(defcard-rg Create-Page
+  "Press button and then search \"test\" "
+  [button-primary {:on-click-fn handler
+                   :label "Create Test Pages and Blocks"}])
+
+
+(defn eval-with-sci
+  [{:keys [eval-str] :as state}]
+  (let [bindings {'athens/db conn
+                  'd/pull d/pull
+                  'd/q d/q
+                  'd/pull-many d/pull-many
+                  'd/entity d/entity}
+        [ok? result] (try
+                       [true (sci/eval-string eval-str {:bindings bindings})]
+                       (catch js/Error e [false e]))]
+    (-> state
+        (assoc :result result)
+        (assoc :error (not ok?)))))
+
+
+(defn eval-box!
+  []
+  (swap! state* eval-with-sci))
+
+
+(defn update-box!
+  [s]
+  (swap! state* assoc :eval-str s))
+
+
+(defn handle-box-change!
+  [e]
+  (update-box! (-> e .-target .-value)))
+
+
+(defn listener
+  [tx-report]
+  (swap! state* update :tx-reports conj tx-report)
+  (when (not (:error @state*))
+    (eval-box!)))
+
+
+(d/listen! conn :devtool listener)
+
+
+(defn handle-shift-return!
+  [e]
+  (.preventDefault e)
+  (eval-box!))
+
+
+(defn insert-tab
+  [s pos]
+  (str (subs s 0 pos) "  " (subs s pos)))
+
+
+(defn handle-tab-key!
+  [e]
+  (let [t (.-target e)
+        v (.-value t)
+        pos (.-selectionStart t)]
+    (.preventDefault e)
+    (update-box! (insert-tab v pos))
+    (set! (.-selectionEnd t) (+ 2 pos))))
+
+
+(defn handle-box-key-down!
+  [e]
+  (let [key-code (.-keyCode e)
+        shift? (.-shiftKey e)
+        k (key-code->key key-code)]
+    (case k
+      :return (when shift?
+                (handle-shift-return! e))
+      :tab (handle-tab-key! e)
+      nil)))
+
+
+(defn error-component
+  [error]
+  [:div {:style {:color "red"}}
+   (str error)])
+
+
+(defn query-component
+  [{:keys [eval-str result error]}]
+  [:div
+   [:textarea {:value eval-str
+               :on-change handle-box-change!
+               :on-key-down handle-box-key-down!
+               :style {:width "100%"
+                       :min-height "150px"
+                       :resize :none
+                       :font-size "12px"
+                       :font-family "IBM Plex Mono"}}]
+   (if-not error
+     [data-browser result]
+     [error-component result])])
+
+
+(defn txes-component
+  [{:keys [tx-reports]}]
+  [data-browser tx-reports])
+
+
+(defn box-component
+  [state _]
+  (let [{:keys [active-panel]} @state
+        switch-panel (fn [panel] (swap! state* assoc :active-panel panel))]
+    [:div
+     [:span [button {:on-click-fn #(switch-panel :query)
+                     :label "Query"}]
+      " "
+      [button {:on-click-fn #(switch-panel :txes)
+               :label "Transactions"}]]
+     (case active-panel
+       :query [query-component @state]
+       :txes [txes-component @state])]))
+
+
+(defcard-rg Reset-to-all-pages
+  (fn []
+    [button {:on-click-fn #(do (swap! state* assoc :eval-str (:eval-str initial-state))
```

Comment:
Shouldn't be too hard to integrate into our application's posh connection, right?

Reply:
Not sure what you mean here, could you explain a bit further?

Reply:
I was wondering what it would look like to integrate this with the main application app, but it should be fairly easy since it's all just dbs.